executor/runtime/docker/docker.go

```diff
@@ -1527,52 +1527,26 @@ const (
 )
 
 func (r *DockerRuntime) setupEFSMounts(parentCtx context.Context, c *runtimeTypes.Container, rootFile *os.File, cred *ucred, efsMountInfos []efsMountInfo) error {
-	baseMountOptions := []string{"vers=4.1,nosharecache,rsize=1048576,wsize=1048576,timeo=600,retrans=2,noresvport"}
-
-	mntNSPath := filepath.Join("/proc", strconv.Itoa(int(cred.pid)), "ns", "mnt")
-	mntNSFile, err := os.OpenFile(mntNSPath, os.O_RDONLY, 0444)
-	if err != nil {
-		return err
-	}
-	defer shouldClose(mntNSFile)
-
-	netNSPath := filepath.Join("/proc", strconv.Itoa(int(cred.pid)), "ns", "net")
-	netNSFile, err := os.OpenFile(netNSPath, os.O_RDONLY, 0444)
-	if err != nil {
-		return err
-	}
-	defer shouldClose(netNSFile)
-
+	baseMountOptions := []string{"vers=4.1,rsize=1048576,wsize=1048576,timeo=600,retrans=2"}
 	for _, efsMountInfo := range efsMountInfos {
 		// Todo: Make into a const
-		// TODO: Run this under the container's PID namespace
 		// Although 5 minutes is probably far too much here, this window is okay to be large
 		// because the parent window should be greater
 		ctx, cancel := context.WithTimeout(parentCtx, 5*time.Minute)
 		defer cancel()
-		cmd := exec.CommandContext(ctx, "/apps/titus-executor/bin/titus-mount") // nolint: gosec
-		// mntNSFD = 3+0 = 3
-		// userNSFD = 3+1 = 4
+		cmd := exec.CommandContext(ctx, "/apps/titus-executor/bin/titus-mount", strconv.Itoa(int(cred.pid))) // nolint: gosec
```

Comment:
Don't you think it is simpler to pass the pid like this? (I assumed so because your version of the code accepted the pid as the first arg, and I thought it was more straightforward than passing the FDs like before)

Reply:
It's simpler, but passing by FD is more "secure" and "reliable" (in that the PID cannot be recycled while in transition). If it's a faff, don't bother, because we're still in tini controlled time here.

Reply:
Yea I would rather not give up the ability to run this command by itself without emulating what titus-executor does.
The pid recycling is real, but should be super rare in this tiny span of time here.